### PR TITLE
feat: Add golang template

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

File: repository.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+templates:
+  - name: TypeScript - Starter
+    path: ./typescript-starter
+  - name: JavaScript - Starter
+    path: ./javascript-starter
+  - name: Golang - Starter
+    path: ./golang-starter
```
Comment (tjholm) [inline repository.yaml:7]: Looks great!

I would hold off officially releasing this for a bit. I would be good to be able to co-ordinate the announcement and release, so we can post to relevant communities/social as well.

Comment (asalkeld) [reply repository.yaml:7]: I have targeted the wrong branch, should be main not develop :-O

**Comment by tjholm** [comment]
We should add a default nitric.yaml file to the templates and stop prompting in the CLI as well.

File: repository.yaml [hunk 2]
```diff
@@ -2,4 +2,6 @@ templates:
   - name: TypeScript - Starter
     path: ./typescript-starter
   - name: JavaScript - Starter
-    path: ./javascript-starter
\ No newline at end of file
+    path: ./javascript-starter
+  - name: Golang - Starter
```
Comment (tjholm) [inline repository.yaml:6]: we should also flag the support state in the template name for golang, to make it clear that that dedicated documentation and support aren't available at this stage (Jye suggested flagging it with community for now)

Comment (asalkeld) [reply repository.yaml:6]: we will need support for that in the cli first

**Pull request closed without merging**